components/scream/src/share/atm_process/atmosphere_diagnostic.hpp

```diff
@@ -38,13 +38,12 @@ class AtmosphereDiagnostic : public AtmosphereProcess
   AtmosphereProcessType type () const { return AtmosphereProcessType::Diagnostic; }
 
   // Getting the diagnostic output
-  Field& get_diagnostic (const Real dt);
+  Field get_diagnostic (const Real dt);
 
+  void set_computed_field (const Field& f);
```

Comment:
Can you mark the methods with `final`, so that derived class cannot override them again?